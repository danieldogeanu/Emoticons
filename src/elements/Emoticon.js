import React from 'react';

const Emoticon = (props) => (
	<span role="img"
		className="Emoticon"
		aria-label={props.name}>
		{props.char}
	</span>
);

export default Emoticon;
