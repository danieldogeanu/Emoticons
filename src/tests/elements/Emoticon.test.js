import React from 'react';
import {render} from '@testing-library/react';
import Emoticon from '../../elements/Emoticon';

describe('Emoticon Element', () => {

	it('renders emoticon element properly', () => {
		const testEmoticon = {
			char: 'ðŸ˜„',
			name: 'grinning face with smiling eyes',
		};
		const {getByRole} = render(
			<Emoticon name={testEmoticon.name} char={testEmoticon.char} />
		);
		const renderedEmoticon = getByRole('img');

		expect(renderedEmoticon).toBeInTheDocument();
		expect(renderedEmoticon).toHaveClass('Emoticon');
		expect(renderedEmoticon).toHaveAttribute('aria-label', testEmoticon.name);
		expect(renderedEmoticon).toHaveTextContent(testEmoticon.char);
		expect(renderedEmoticon).toMatchSnapshot();
	});

});
