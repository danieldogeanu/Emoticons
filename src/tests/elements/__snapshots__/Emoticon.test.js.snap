// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Emoticon Element renders emoticon element properly 1`] = `
<span
  aria-label="grinning face with smiling eyes"
  class="Emoticon"
  role="img"
>
  ðŸ˜„
</span>
`;
